pkg/cloud/services/compute/instance.go

```diff
@@ -113,6 +104,25 @@ func (s *Service) CreateInstance(openStackCluster *infrav1.OpenStackCluster, mac
 	if err != nil {
 		return nil, err
 	}
+	// verify trunk support if trunk is set on at least one Port.
+	for _, net := range nets {
+		port := net.PortOpts
+		if port != nil {
+			// No Trunk field specified for the port, inherit openStackMachine.Spec.Trunk.
+			if port.Trunk == nil {
+				*port.Trunk = openStackMachine.Spec.Trunk
+			}
+			if openStackMachine.Spec.Trunk || *port.Trunk {
```

Comment:
There is nowhere else where `openStackMachine.Spec.Trunk` is checked. So, we need both.

Reply:
What we're saying here is that if Trunk is set on the Machine it will be set on all ports, regardless of the explicit setting for that port. Is that correct?

Reply:
The opposite:

1. If set@port => take port level and ignore Machine level
2. if not set@port => take Machine level

on verifying trunk support:
(a) No matter how trunk was setup via (1) or (2):  check at port level `(*port.Trunk)`
(b) In case no port was supplied => check at instance level` (openStackMachine.Spec.Trunk)`